sentinel.go

```diff
@@ -468,9 +473,17 @@ func (sc *Sentinel) innerSpin() error {
 			// loop
 		case <-sc.pconnCh:
 			switchMaster = true
+			if waitFor := atomic.SwapUint32(&sc.testSleepBeforeSwitch, 0); waitFor > 0 {
```

Comment:
@mediocregopher this is kinda hacky but I couldn't think about a better way for this